cve_bin_tool/helper_script.py

```diff
@@ -83,35 +83,44 @@ def extract_and_parse_file(self, filename):
                                 self.filename_pattern.append(filepath.split("\\")[-1])
                             else:
                                 self.filename_pattern.append(filepath.split("/")[-1])
+                            LOGGER.info(
+                                f"matches for {self.product} found in {clean_path}"
+                            )
+
                             binary_string_list += matches
 
                             for i in matches:
-                                if ("/" not in i and "!" not in i) and len(i) > 40:
+                                if ("/" not in i and "!" not in i) and len(
+                                    i
+                                ) > self.string_length:
                                     self.contain_patterns.append(i)
 
-                        # self.logger.debug(f"{self.filename_pattern}")
+                        LOGGER.debug(f"{self.filename_pattern}")
 
                 # to resolve case when there are no strings common with product_name in them
                 if self.contain_patterns:
                     return self.contain_patterns
-                else:
-                    return binary_string_list
+                return binary_string_list
 
-    def search_pattern(self, string_list):
+    def search_pattern(self, file_content, pattern):
         """find strings for CONTAIN_PATTERNS with product_name in them"""
 
-        pattern = rf"{self.product}"
-        matched = [
-            i.strip() for i in string_list if re.search(pattern, i, re.IGNORECASE)
+        file_content_list = file_content.split()
+        matches = [
+            i.strip() for i in file_content_list if re.search(pattern, i, re.IGNORECASE)
         ]
-        return matched
+        LOGGER.debug(f"{matches}")  # TODO: regex highlight in these matched strings?
+        return matches
 
     def search_version_string(self, matched_list):
         """finds version strings from matched list"""
 
-        # pattern = rf"{self.product}(.*){self.version}"
+        # TODO: add multiline string finding
+
+        pattern1 = rf"{self.product}(.*){self.version}"
         # ^ this does not work for debian packages
-        pattern = rf"{self.product}(.*)([0-9]+[.-][0-9]+([.-][0-9]+)?)"
+
+        # pattern2 = rf"{self.product}(.*)([0-9]+[.-][0-9]+([.-][0-9]+)?)"
```

Comment:
Now, I'm using `pattern1` instead of `pattern2` for more accurate version-string finding. But, `pattern1` would require the user to manually specify the `--version` for some of the debian and bsd packages. But, if the version is specified and if a single line version string exists (currently helper-script does not support multiline regex finding), then the `VERSION_PATTERNS` is much more accurate. Is it a good trade-off?